cxplat/src/cxplat_winkernel/module_winkernel.c

```diff
@@ -21,34 +22,43 @@ cxplat_convert_ansi_to_utf8_string(_In_ ANSI_STRING* ansi_string, _Out_ cxplat_u
         goto Exit;
     }
 
-    // Convert the unicode string to a UTF-8 string.
-    status = RtlUnicodeStringToUTF8String(&utf8_str, &unicode_str, TRUE);
+    // Calculate the required UTF-8 buffer size by calling RtlUnicodeToUTF8N with NULL buffer
+    status = RtlUnicodeToUTF8N(NULL, 0, &utf8_buffer_size, unicode_str.Buffer, unicode_str.Length);
     if (!NT_SUCCESS(status)) {
```

Comment:
the exit path looks to see if utf8->value is null. but we don't initialize that out param. so we could end up trying to free garbage. can initialize the OUT params at the start of the function to guarantee a known state?